pandas/core/internals/construction.py

```diff
@@ -142,7 +137,7 @@ def arrays_to_mgr(
 
 
 def rec_array_to_mgr(
-    data: MaskedRecords | np.recarray | np.ndarray,
+    data: np.ma.MaskedArray | np.recarray | np.ndarray,
```

Comment:
why this change?

Reply:
There's a 
```
    # fill if needed
    if isinstance(data, np.ma.MaskedArray):
        new_arrays = fill_masked_arrays(data, arr_columns)
```
in the body that implies that MaskedArray (which is a superclass?) is allowed?

Reply:
In all tests that get here, we have `isinstance(data, np.ma.mrecords.MaskedRecords)`.  AFAICT the check is for MaskedArray bc MaskedRecords requires a separate import that we are avoiding at import-time

Reply:
`MaskedRecords`, `MaskedArray` and `recarray` are all subclass of `np.ndarray` so the annotation here is redundant and should just be `np.ndarray`

Since `MaskedArray` is a subclass of `np.ndarray`, it is a valid type to pass to `rec_array_to_mgr` and mypy is correct that we pass off `MaskedArray` classes and subclasses to `fill_masked_arrays`

> In all tests that get here

Isn't the point of mypy to catch the cases where we don't have tests?

but ok to leave for now. as we now have a cast and comment to highlight the issue